Hanging up when installing mod

**Describe the bug**
The game doesn't finish installing a mod.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to the mods page
2. Search for "Deep"
3. Click on Deep Pockets and install
4. See that it gets stuck

**Expected behavior**
The mod should get installed in a reasonable amount of time.

**Please complete the following information:**
 - OS: Ubuntu 20.04
 - Minecraft version 1.17.1
 - Version 1.1.1-alpha+1.17

**Logs**
Latest log:
<details>
<summary>Log (click to extend)</summary>

MultiMC version: 0.6.13-2519
Mods:
  [✔️] cloth-config-5.0.38-fabric
  [✔️] fabric-api-0.40.8+1.17
  [✔️] fabric-language-kotlin-1.6.5+kotlin.1.5.31
  [✔️] modmanager-1.1.1-alpha+1.17
  [✔️] modmenu-2.0.14
  [✔️] paintings-fabric-1.17.1-1.0.0.3

Params:
  --username  --version MultiMC5 --gameDir /home/tim/.local/share/multimc/instances/mods you should know 1.17/.minecraft --assetsDir /home/tim/.local/share/multimc/assets --assetIndex 1.17 --uuid  --accessToken  --userType  --versionType release

Window size: max (if available)

Java Arguments:
[-Xms1024m, -Xmx4096m, -Duser.language=en]


Minecraft process ID: 47826


Using onesix launcher.

[13:48:38] [main/INFO]: Loading for game Minecraft 1.17.1
[13:48:38] [main/INFO]: Loading 62 mods:
	- cloth-basic-math 0.5.1
	- cloth-config2 5.0.38
	- fabric 0.40.8+1.17
	- fabric-api-base 0.3.0+a02b446318
	- fabric-api-lookup-api-v1 1.3.0+cbda931818
	- fabric-biome-api-v1 3.2.0+cbda931818
	- fabric-blockrenderlayer-v1 1.1.5+a02b446318
	- fabric-command-api-v1 1.1.3+5ab9934c18
	- fabric-commands-v0 0.2.2+92519afa18
	- fabric-containers-v0 0.1.12+cbda931818
	- fabric-content-registries-v0 0.3.0+cbda931818
	- fabric-crash-report-info-v1 0.1.5+be9da31018
	- fabric-dimensions-v1 2.0.11+6cefd57718
	- fabric-entity-events-v1 1.2.3+87cc6e4c18
	- fabric-events-interaction-v0 0.4.10+fc40aa9d18
	- fabric-events-lifecycle-v0 0.2.1+92519afa18
	- fabric-game-rule-api-v1 1.0.7+cbda931818
	- fabric-item-api-v1 1.2.4+cbda931818
	- fabric-item-groups-v0 0.2.10+b7ab612118
	- fabric-key-binding-api-v1 1.0.4+cbda931818
	- fabric-keybindings-v0 0.2.2+36b77c3e18
	- fabric-language-kotlin 1.6.5+kotlin.1.5.31
	- fabric-lifecycle-events-v1 1.4.4+a02b446318
	- fabric-loot-tables-v1 1.0.4+a02b446318
	- fabric-mining-levels-v0 0.1.3+92519afa18
	- fabric-models-v0 0.3.0+a02b446318
	- fabric-networking-api-v1 1.0.13+cbda931818
	- fabric-networking-blockentity-v0 0.2.11+a02b446318
	- fabric-networking-v0 0.3.2+92519afa18
	- fabric-object-builder-api-v1 1.10.9+cbda931818
	- fabric-object-builders-v0 0.7.3+a02b446318
	- fabric-particles-v1 0.2.4+a02b446318
	- fabric-registry-sync-v0 0.7.11+7931163218
	- fabric-renderer-api-v1 0.4.4+cbda931818
	- fabric-renderer-indigo 0.4.8+cbda931818
	- fabric-renderer-registries-v1 3.2.4+7931163218
	- fabric-rendering-data-attachment-v1 0.1.5+a02b446318
	- fabric-rendering-fluids-v1 0.1.14+4658223018
	- fabric-rendering-v0 1.1.5+7931163218
	- fabric-rendering-v1 1.9.0+7931163218
	- fabric-resource-loader-v0 0.4.8+a00e834b18
	- fabric-screen-api-v1 1.0.4+cbda931818
	- fabric-screen-handler-api-v1 1.1.8+cbda931818
	- fabric-structure-api-v1 1.1.13+5ab9934c18
	- fabric-tag-extensions-v0 1.2.1+b06cb95b18
	- fabric-textures-v0 1.0.6+a02b446318
	- fabric-tool-attribute-api-v1 1.2.12+b7ab612118
	- fabric-transfer-api-v1 1.4.0+7931163218
	- fabricloader 0.12.1
	- java 16
	- minecraft 1.17.1
	- modmanager 1.1.1-alpha+1.17
	- modmenu 2.0.14
	- org_jetbrains_kotlin_kotlin-reflect 1.5.31
	- org_jetbrains_kotlin_kotlin-stdlib 1.5.31
	- org_jetbrains_kotlin_kotlin-stdlib-jdk7 1.5.31
	- org_jetbrains_kotlin_kotlin-stdlib-jdk8 1.5.31
	- org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm 1.5.2
	- org_jetbrains_kotlinx_kotlinx-coroutines-jdk8 1.5.2
	- org_jetbrains_kotlinx_kotlinx-serialization-core-jvm 1.3.0
	- org_jetbrains_kotlinx_kotlinx-serialization-json-jvm 1.3.0
	- paintings 1.17.1-1.0.0.3
[13:48:38] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.4 Source=file:/home/tim/.local/share/multimc/libraries/net/fabricmc/sponge-mixin/0.10.2+mixin.0.8.4/sponge-mixin-0.10.2+mixin.0.8.4.jar Service=Knot/Fabric Env=CLIENT
[13:48:38] [main/INFO]: Compatibility level set to JAVA_16
[13:48:45] [Render thread/INFO]: Environment: authHost='https://authserver.mojang.com', accountsHost='https://api.mojang.com', sessionHost='https://sessionserver.mojang.com', servicesHost='https://api.minecraftservices.com', name='PROD'
[13:48:46] [Render thread/INFO]: Setting user: lTeaJayl
[13:48:46] [Render thread/WARN]: @Inject(@At("INVOKE")) Shift.BY=3 on fabric-lifecycle-events-v1.mixins.json:client.WorldChunkMixin::handler$zdc000$onLoadBlockEntity exceeds the maximum allowed value: 0. Increase the value of maxShiftBy to suppress this warning.
[13:48:46] [Render thread/INFO]: loading json file and contents for paintings.
[13:48:46] [Render thread/INFO]: Copying Over Base Template to /paintings
[13:48:46] [Render thread/INFO]: Reading out ResourcePacks to find painting related json files
[13:48:46] [Render thread/INFO]: Started Reading all json files in /painting directory
[13:48:46] [Render thread/INFO]: ./paintings/paintings.json
[13:48:46] [Render thread/INFO]: Loaded json painting abstract_blue , 16 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting abstract_rainbow , 16 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting abstract_red , 16 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting abstract_sunset , 16 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting arachnophobe , 32 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting barn_owl , 32 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting big_z , 16 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting blue_bird , 32 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting bluesclues , 32 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting borgia , 16 x 32
[13:48:46] [Render thread/INFO]: Loaded json painting cane , 16 x 32
[13:48:46] [Render thread/INFO]: Loaded json painting cat_black , 16 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting cat_gray , 16 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting cat_orange , 16 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting cat , 32 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting colorful_squares , 32 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting crest , 32 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting danger_zone , 32 x 32
[13:48:46] [Render thread/INFO]: Loaded json painting decorative_gun , 32 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting exit_down , 32 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting exit_up , 32 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting exit_left , 32 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting exit_right , 32 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting face_bat , 16 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting face_chicken , 16 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting face_cow , 16 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting face_creeper , 16 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting face_dog , 16 x 16
[13:48:46] [Render thread/INFO]: Loaded json painting face_enderman , 16 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting face_pig , 16 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting face_pigman , 16 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting face_silverfish , 16 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting face_skeleton , 16 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting face_squid , 16 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting face_zombie , 16 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting fishes , 32 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting flowers , 32 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting fruits , 32 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting ghost , 16 x 32
[13:48:47] [Render thread/INFO]: Loaded json painting glowlamp , 32 x 32
[13:48:47] [Render thread/INFO]: Loaded json painting glowstone_hourglass , 16 x 32
[13:48:47] [Render thread/INFO]: Loaded json painting iluvmc , 32 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting link , 32 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting mine_prosperity , 32 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting no_trespassing_for_mobs , 32 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting ocelot , 16 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting penguin , 32 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting pig_on_a_cliff , 32 x 32
[13:48:47] [Render thread/INFO]: Loaded json painting pkmn_blue , 32 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting pkmn_red , 32 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting pkmn_green , 32 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting plains_hut , 32 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting portrait_2 , 16 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting portrait , 32 x 32
[13:48:47] [Render thread/INFO]: Loaded json painting prison , 16 x 32
[13:48:47] [Render thread/INFO]: Loaded json painting prosperity , 16 x 32
[13:48:47] [Render thread/INFO]: Loaded json painting rest , 32 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting skeleton , 16 x 32
[13:48:47] [Render thread/INFO]: Loaded json painting sky , 32 x 32
[13:48:47] [Render thread/INFO]: Loaded json painting skyblock , 32 x 32
[13:48:47] [Render thread/INFO]: Loaded json painting snake , 16 x 32
[13:48:47] [Render thread/INFO]: Loaded json painting snow_landscape , 32 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting subaraki , 32 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting synth_city , 16 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting tapistry_a , 32 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting tapistry_b , 16 x 32
[13:48:47] [Render thread/INFO]: Loaded json painting tapistry_purple , 16 x 32
[13:48:47] [Render thread/INFO]: Loaded json painting torched , 32 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting waterfall , 32 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting whale , 32 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting wheat_field , 32 x 16
[13:48:47] [Render thread/INFO]: Loaded json painting wolf_in_wheat , 32 x 16
[13:48:47] [Render thread/INFO]: Registered painting paintings:abstract_blue
[13:48:47] [Render thread/INFO]: Registered painting paintings:abstract_rainbow
[13:48:47] [Render thread/INFO]: Registered painting paintings:abstract_red
[13:48:47] [Render thread/INFO]: Registered painting paintings:abstract_sunset
[13:48:47] [Render thread/INFO]: Registered painting paintings:arachnophobe
[13:48:47] [Render thread/INFO]: Registered painting paintings:barn_owl
[13:48:47] [Render thread/INFO]: Registered painting paintings:big_z
[13:48:47] [Render thread/INFO]: Registered painting paintings:blue_bird
[13:48:47] [Render thread/INFO]: Registered painting paintings:bluesclues
[13:48:47] [Render thread/INFO]: Registered painting paintings:borgia
[13:48:47] [Render thread/INFO]: Registered painting paintings:cane
[13:48:47] [Render thread/INFO]: Registered painting paintings:cat_black
[13:48:47] [Render thread/INFO]: Registered painting paintings:cat_gray
[13:48:47] [Render thread/INFO]: Registered painting paintings:cat_orange
[13:48:47] [Render thread/INFO]: Registered painting paintings:cat
[13:48:47] [Render thread/INFO]: Registered painting paintings:colorful_squares
[13:48:47] [Render thread/INFO]: Registered painting paintings:crest
[13:48:47] [Render thread/INFO]: Registered painting paintings:danger_zone
[13:48:47] [Render thread/INFO]: Registered painting paintings:decorative_gun
[13:48:47] [Render thread/INFO]: Registered painting paintings:exit_down
[13:48:47] [Render thread/INFO]: Registered painting paintings:exit_up
[13:48:47] [Render thread/INFO]: Registered painting paintings:exit_left
[13:48:47] [Render thread/INFO]: Registered painting paintings:exit_right
[13:48:47] [Render thread/INFO]: Registered painting paintings:face_bat
[13:48:47] [Render thread/INFO]: Registered painting paintings:face_chicken
[13:48:47] [Render thread/INFO]: Registered painting paintings:face_cow
[13:48:47] [Render thread/INFO]: Registered painting paintings:face_creeper
[13:48:47] [Render thread/INFO]: Registered painting paintings:face_dog
[13:48:47] [Render thread/INFO]: Registered painting paintings:face_enderman
[13:48:47] [Render thread/INFO]: Registered painting paintings:face_pig
[13:48:47] [Render thread/INFO]: Registered painting paintings:face_pigman
[13:48:47] [Render thread/INFO]: Registered painting paintings:face_silverfish
[13:48:47] [Render thread/INFO]: Registered painting paintings:face_skeleton
[13:48:47] [Render thread/INFO]: Registered painting paintings:face_squid
[13:48:47] [Render thread/INFO]: Registered painting paintings:face_zombie
[13:48:47] [Render thread/INFO]: Registered painting paintings:fishes
[13:48:47] [Render thread/INFO]: Registered painting paintings:flowers
[13:48:47] [Render thread/INFO]: Registered painting paintings:fruits
[13:48:47] [Render thread/INFO]: Registered painting paintings:ghost
[13:48:47] [Render thread/INFO]: Registered painting paintings:glowlamp
[13:48:47] [Render thread/INFO]: Registered painting paintings:glowstone_hourglass
[13:48:47] [Render thread/INFO]: Registered painting paintings:iluvmc
[13:48:47] [Render thread/INFO]: Registered painting paintings:link
[13:48:47] [Render thread/INFO]: Registered painting paintings:mine_prosperity
[13:48:47] [Render thread/INFO]: Registered painting paintings:no_trespassing_for_mobs
[13:48:47] [Render thread/INFO]: Registered painting paintings:ocelot
[13:48:47] [Render thread/INFO]: Registered painting paintings:penguin
[13:48:47] [Render thread/INFO]: Registered painting paintings:pig_on_a_cliff
[13:48:47] [Render thread/INFO]: Registered painting paintings:pkmn_blue
[13:48:47] [Render thread/INFO]: Registered painting paintings:pkmn_red
[13:48:47] [Render thread/INFO]: Registered painting paintings:pkmn_green
[13:48:47] [Render thread/INFO]: Registered painting paintings:plains_hut
[13:48:47] [Render thread/INFO]: Registered painting paintings:portrait_2
[13:48:47] [Render thread/INFO]: Registered painting paintings:portrait
[13:48:47] [Render thread/INFO]: Registered painting paintings:prison
[13:48:47] [Render thread/INFO]: Registered painting paintings:prosperity
[13:48:47] [Render thread/INFO]: Registered painting paintings:rest
[13:48:47] [Render thread/INFO]: Registered painting paintings:skeleton
[13:48:47] [Render thread/INFO]: Registered painting paintings:sky
[13:48:47] [Render thread/INFO]: Registered painting paintings:skyblock
[13:48:47] [Render thread/INFO]: Registered painting paintings:snake
[13:48:47] [Render thread/INFO]: Registered painting paintings:snow_landscape
[13:48:47] [Render thread/INFO]: Registered painting paintings:subaraki
[13:48:47] [Render thread/INFO]: Registered painting paintings:synth_city
[13:48:47] [Render thread/INFO]: Registered painting paintings:tapistry_a
[13:48:47] [Render thread/INFO]: Registered painting paintings:tapistry_b
[13:48:47] [Render thread/INFO]: Registered painting paintings:tapistry_purple
[13:48:47] [Render thread/INFO]: Registered painting paintings:torched
[13:48:47] [Render thread/INFO]: Registered painting paintings:waterfall
[13:48:47] [Render thread/INFO]: Registered painting paintings:whale
[13:48:47] [Render thread/INFO]: Registered painting paintings:wheat_field
[13:48:47] [Render thread/INFO]: Registered painting paintings:wolf_in_wheat
[13:48:47] [Render thread/INFO]: [Indigo] Registering Indigo renderer!
[13:48:49] [DefaultDispatcher-worker-1/INFO]: Failed to read file ignoring it: Input length = 3
[13:48:49] [DefaultDispatcher-worker-1/INFO]: Failed to read file ignoring it: Input length = 1
[13:48:49] [DefaultDispatcher-worker-1/INFO]: Failed to read file ignoring it: Input length = 1
[13:48:49] [DefaultDispatcher-worker-1/INFO]: Failed to read file ignoring it: Input length = 1
[13:48:49] [DefaultDispatcher-worker-1/INFO]: Failed to read file ignoring it: Input length = 1
[13:48:49] [DefaultDispatcher-worker-1/INFO]: Failed to read file ignoring it: Input length = 1
[13:48:49] [Render thread/WARN]: WARNING! Mod cloth-basic-math is only using deprecated 'modmenu:api' custom value! This will be removed in 1.18 snapshots, so ask the author of this mod to support the new API.
[13:48:49] [DefaultDispatcher-worker-3/INFO]: Searching for updates for cloth-config2 using fallback method
[13:48:49] [DefaultDispatcher-worker-2/INFO]: Skipping update for cloth-basic-math because it has no jar in mods
[13:48:49] [DefaultDispatcher-worker-4/INFO]: Searching for updates for modmanager using defined mod id
[13:48:49] [DefaultDispatcher-worker-2/INFO]: Searching for updates for modmenu using fallback method
[13:48:49] [DefaultDispatcher-worker-5/INFO]: Searching for updates for paintings using fallback method
[13:48:50] [Render thread/INFO]: Backend library: LWJGL version 3.2.2 build 10
[13:48:50] [DefaultDispatcher-worker-3/WARN]: Error while searching for fallback id for mod cloth-config2: No possible match found
[13:48:50] [DefaultDispatcher-worker-5/WARN]: Error while searching for fallback id for mod paintings: No possible match found
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: java.lang.NoClassDefFoundError: net/fabricmc/loader/util/version/VersionDeserializer
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: 	at xyz.deathsgun.modmanager.update.VersionFinder.findUpdateByVersion$modmanager(VersionFinder.kt:55)
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: 	at xyz.deathsgun.modmanager.update.VersionFinder.findUpdate(VersionFinder.kt:81)
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: 	at xyz.deathsgun.modmanager.update.UpdateManager.checkForUpdates(UpdateManager.kt:200)
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: 	at xyz.deathsgun.modmanager.update.UpdateManager.access$checkForUpdates(UpdateManager.kt:57)
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: 	at xyz.deathsgun.modmanager.update.UpdateManager$checkUpdates$2$1$1.invokeSuspend(UpdateManager.kt:81)
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: 	Suppressed: java.lang.NoClassDefFoundError: net/fabricmc/loader/util/version/VersionDeserializer
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: 		at xyz.deathsgun.modmanager.update.VersionFinder.findUpdateByVersion$modmanager(VersionFinder.kt:55)
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: 		at xyz.deathsgun.modmanager.update.VersionFinder.findUpdate(VersionFinder.kt:81)
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: 		at xyz.deathsgun.modmanager.update.UpdateManager.checkForUpdatesManually(UpdateManager.kt:154)
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: 		at xyz.deathsgun.modmanager.update.UpdateManager.access$checkForUpdatesManually(UpdateManager.kt:57)
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: 		at xyz.deathsgun.modmanager.update.UpdateManager$checkUpdates$2$1$1.invokeSuspend(UpdateManager.kt:77)
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: 		... 6 more
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: Caused by: java.lang.ClassNotFoundException: net.fabricmc.loader.util.version.VersionDeserializer
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: 	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:636)
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: 	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:182)
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:519)
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: 	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:158)
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:519)
[13:48:51] [DefaultDispatcher-worker-4/INFO]: [STDERR]: 	... 11 more
[13:48:52] [Render thread/INFO]: Narrator library successfully loaded
[13:48:53] [Render thread/INFO]: Reloading ResourceManager: Default, Fabric Mods (Cloth Config v5, Fabric API, Fabric API Base, Fabric API Lookup API (v1), Fabric Biome API (v1), Fabric BlockRenderLayer Registration (v1), Fabric Command API (v1), Fabric Commands (v0), Fabric Containers (v0), Fabric Content Registries (v0), Fabric Crash Report Info (v1), Fabric Dimensions API (v1), Fabric Entity Events (v1), Fabric Events Interaction (v0), Fabric Events Lifecycle (v0), Fabric Game Rule API (v1), Fabric Item API (v1), Fabric Item Groups (v0), Fabric Key Binding API (v1), Fabric Key Bindings (v0), Fabric Language Kotlin, Fabric Lifecycle Events (v1), Fabric Loot Tables (v1), Fabric Mining Levels (v0), Fabric Models (v0), Fabric Networking API (v1), Fabric Networking Block Entity (v0), Fabric Networking (v0), Fabric Object Builder API (v1), Fabric Object Builders (v0), Fabric Particles (v1), Fabric Registry Sync (v0), Fabric Renderer API (v1), Fabric Renderer - Indigo, Fabric Renderer Registries (v1), Fabric Rendering Data Attachment (v1), Fabric Rendering Fluids (v1), Fabric Rendering (v0), Fabric Rendering (v1), Fabric Resource Loader (v0), Fabric Screen API (v1), Fabric Screen Handler API (v1), Fabric Structure API (v1), Fabric Tag Extensions (v0), Fabric Textures (v0), Fabric Tool Attribute API (v1), Fabric Transfer API (v1), Fabric Loader, Mod Manager, Mod Menu, Paintings ++)
[13:48:57] [Render thread/INFO]: OpenAL initialized.
[13:48:57] [Render thread/INFO]: Sound engine started
[13:48:58] [Render thread/INFO]: Created: 1024x1024x4 minecraft:textures/atlas/blocks.png-atlas
[13:48:58] [Render thread/INFO]: Created: 256x128x4 minecraft:textures/atlas/signs.png-atlas
[13:48:58] [Render thread/INFO]: Created: 1024x512x4 minecraft:textures/atlas/banner_patterns.png-atlas
[13:48:58] [Render thread/INFO]: Created: 1024x512x4 minecraft:textures/atlas/shield_patterns.png-atlas
[13:48:58] [Render thread/INFO]: Created: 256x256x4 minecraft:textures/atlas/chest.png-atlas
[13:48:58] [Render thread/INFO]: Created: 512x256x4 minecraft:textures/atlas/beds.png-atlas
[13:48:58] [Render thread/INFO]: Created: 512x256x4 minecraft:textures/atlas/shulker_boxes.png-atlas
[13:49:01] [Render thread/INFO]: Created: 256x256x0 minecraft:textures/atlas/particles.png-atlas
[13:49:01] [Render thread/INFO]: Created: 512x256x0 minecraft:textures/atlas/paintings.png-atlas
[13:49:01] [Render thread/INFO]: Created: 256x128x0 minecraft:textures/atlas/mob_effects.png-atlas
[13:49:59] [DefaultDispatcher-worker-6/INFO]: Installing DeepslateCutting Fabric
[13:50:00] [DefaultDispatcher-worker-6/INFO]: [STDERR]: java.lang.NoClassDefFoundError: net/fabricmc/loader/util/version/VersionDeserializer
[13:50:00] [DefaultDispatcher-worker-6/INFO]: [STDERR]: 	at xyz.deathsgun.modmanager.update.VersionFinder.findUpdateByVersion$modmanager(VersionFinder.kt:55)
[13:50:00] [DefaultDispatcher-worker-6/INFO]: [STDERR]: 	at xyz.deathsgun.modmanager.update.VersionFinder.findUpdate(VersionFinder.kt:81)
[13:50:00] [DefaultDispatcher-worker-6/INFO]: [STDERR]: 	at xyz.deathsgun.modmanager.update.UpdateManager.installMod(UpdateManager.kt:234)
[13:50:00] [DefaultDispatcher-worker-6/INFO]: [STDERR]: 	at xyz.deathsgun.modmanager.gui.ModProgressScreen.installMod(ModProgressScreen.kt:120)
[13:50:00] [DefaultDispatcher-worker-6/INFO]: [STDERR]: 	at xyz.deathsgun.modmanager.gui.ModProgressScreen.access$installMod(ModProgressScreen.kt:40)
[13:50:00] [DefaultDispatcher-worker-6/INFO]: [STDERR]: 	at xyz.deathsgun.modmanager.gui.ModProgressScreen$init$2.invokeSuspend(ModProgressScreen.kt:71)
[13:50:00] [DefaultDispatcher-worker-6/INFO]: [STDERR]: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
[13:50:00] [DefaultDispatcher-worker-6/INFO]: [STDERR]: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
[13:50:00] [DefaultDispatcher-worker-6/INFO]: [STDERR]: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
[13:50:00] [DefaultDispatcher-worker-6/INFO]: [STDERR]: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
[13:50:00] [DefaultDispatcher-worker-6/INFO]: [STDERR]: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
[13:50:00] [DefaultDispatcher-worker-6/INFO]: [STDERR]: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
[13:50:11] [DefaultDispatcher-worker-6/INFO]: Installing DeepslateCutting Fabric
[13:50:11] [DefaultDispatcher-worker-1/INFO]: Installing DeepslateCutting Fabric
[13:50:11] [DefaultDispatcher-worker-6/INFO]: [STDERR]: java.lang.NoClassDefFoundError: net/fabricmc/loader/util/version/VersionDeserializer
[13:50:11] [DefaultDispatcher-worker-6/INFO]: [STDERR]: 	at xyz.deathsgun.modmanager.update.VersionFinder.findUpdateByVersion$modmanager(VersionFinder.kt:55)
[13:50:11] [DefaultDispatcher-worker-6/INFO]: [STDERR]: 	at xyz.deathsgun.modmanager.update.VersionFinder.findUpdate(VersionFinder.kt:81)
[13:50:11] [DefaultDispatcher-worker-6/INFO]: [STDERR]: 	at xyz.deathsgun.modmanager.update.UpdateManager.installMod(UpdateManager.kt:234)
[13:50:11] [DefaultDispatcher-worker-6/INFO]: [STDERR]: 	at xyz.deathsgun.modmanager.gui.ModProgressScreen.installMod(ModProgressScreen.kt:120)
[13:50:11] [DefaultDispatcher-worker-6/INFO]: [STDERR]: 	at xyz.deathsgun.modmanager.gui.ModProgressScreen.access$installMod(ModProgressScreen.kt:40)
[13:50:11] [DefaultDispatcher-worker-6/INFO]: [STDERR]: 	at xyz.deathsgun.modmanager.gui.ModProgressScreen$init$2.invokeSuspend(ModProgressScreen.kt:71)
[13:50:11] [DefaultDispatcher-worker-6/INFO]: [STDERR]: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
[13:50:11] [DefaultDispatcher-worker-6/INFO]: [STDERR]: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
[13:50:11] [DefaultDispatcher-worker-6/INFO]: [STDERR]: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
[13:50:11] [DefaultDispatcher-worker-6/INFO]: [STDERR]: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
[13:50:11] [DefaultDispatcher-worker-6/INFO]: [STDERR]: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
[13:50:11] [DefaultDispatcher-worker-6/INFO]: [STDERR]: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
[13:50:11] [DefaultDispatcher-worker-1/INFO]: [STDERR]: java.lang.NoClassDefFoundError: net/fabricmc/loader/util/version/VersionDeserializer
[13:50:11] [DefaultDispatcher-worker-1/INFO]: [STDERR]: 	at xyz.deathsgun.modmanager.update.VersionFinder.findUpdateByVersion$modmanager(VersionFinder.kt:55)
[13:50:11] [DefaultDispatcher-worker-1/INFO]: [STDERR]: 	at xyz.deathsgun.modmanager.update.VersionFinder.findUpdate(VersionFinder.kt:81)
[13:50:11] [DefaultDispatcher-worker-1/INFO]: [STDERR]: 	at xyz.deathsgun.modmanager.update.UpdateManager.installMod(UpdateManager.kt:234)
[13:50:11] [DefaultDispatcher-worker-1/INFO]: [STDERR]: 	at xyz.deathsgun.modmanager.gui.ModProgressScreen.installMod(ModProgressScreen.kt:120)
[13:50:11] [DefaultDispatcher-worker-1/INFO]: [STDERR]: 	at xyz.deathsgun.modmanager.gui.ModProgressScreen.access$installMod(ModProgressScreen.kt:40)
[13:50:11] [DefaultDispatcher-worker-1/INFO]: [STDERR]: 	at xyz.deathsgun.modmanager.gui.ModProgressScreen$init$2.invokeSuspend(ModProgressScreen.kt:71)
[13:50:11] [DefaultDispatcher-worker-1/INFO]: [STDERR]: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
[13:50:11] [DefaultDispatcher-worker-1/INFO]: [STDERR]: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
[13:50:11] [DefaultDispatcher-worker-1/INFO]: [STDERR]: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
[13:50:11] [DefaultDispatcher-worker-1/INFO]: [STDERR]: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
[13:50:11] [DefaultDispatcher-worker-1/INFO]: [STDERR]: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
[13:50:11] [DefaultDispatcher-worker-1/INFO]: [STDERR]: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
[13:50:15] [Render thread/INFO]: Stopping!
Exception in thread "DefaultDispatcher-worker-4" Exception in thread "DefaultDispatcher-worker-6" Exception in thread "DefaultDispatcher-worker-6" Exception in thread "DefaultDispatcher-worker-1" 
Process exited with code 0.
Clipboard copy at: 15 Oct 2021 13:51:56 +0200

</details>

--- a/gradle.properties
+++ b/gradle.properties
@@ -13,15 +13,13 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-
 # Done to increase the memory available to gradle.
 org.gradle.jvmargs=-Xms2G -Xmx4G
-
 # Fabric Properties
 # check this on https://modmuss50.me/fabric.html
 minecraft_version=1.17.1
-yarn_mappings=1.17.1+build.39
-loader_version=0.11.6
+yarn_mappings=1.17.1+build.61
+loader_version=0.12.1
 # Mod Properties
 mod_version=1.1.1-alpha+1.17
 maven_group=xyz.deathsgun

--- a/src/main/kotlin/xyz/deathsgun/modmanager/update/VersionFinder.kt
+++ b/src/main/kotlin/xyz/deathsgun/modmanager/update/VersionFinder.kt
@@ -18,7 +18,6 @@ package xyz.deathsgun.modmanager.update
 
 import net.fabricmc.loader.api.SemanticVersion
 import net.fabricmc.loader.api.VersionParsingException
-import net.fabricmc.loader.util.version.VersionDeserializer
 import xyz.deathsgun.modmanager.api.mod.Version
 import xyz.deathsgun.modmanager.config.Config
 
@@ -52,9 +51,9 @@ object VersionFinder {
             .filter { it.gameVersions.any { it1 -> it1.startsWith(mcVersion) } }
         var latestVersion: Version? = null
         var latestVer: SemanticVersion? = null
-        val installedVer = VersionDeserializer.deserializeSemantic(installedVersion)
+        val installedVer = SemanticVersion.parse(installedVersion)
         for (version in versions) {
-            val parsedVersion = VersionDeserializer.deserializeSemantic(version.version)
+            val parsedVersion = SemanticVersion.parse(version.version)
             if (latestVersion == null) {
                 latestVersion = version
                 latestVer = parsedVersion

--- a/src/main/resources/fabric.mod.json
+++ b/src/main/resources/fabric.mod.json
@@ -35,7 +35,7 @@
     "modmanager.mixins.json"
   ],
   "depends": {
-    "fabricloader": ">=0.11.3",
+    "fabricloader": ">=0.12",
     "modmenu": ">=${modmenu_version}",
     "fabric-language-kotlin": ">=${fabric_kotlin_version}",
     "minecraft": "1.17.x",

